src/dramtypes/DRAMPowerDDR4.cc

```diff
@@ -0,0 +1,492 @@
+                counters.handlePupPre(timestamp);
+            } else if (type == MemCommand::SREN) {
+                counters.handleSREn(bank, timestamp);
+            } else if (type == MemCommand::SREX) {
+                counters.handleSREx(bank, timestamp);
+            } else if (type == MemCommand::END || type == MemCommand::NOP) {
+                counters.handleNopEnd(timestamp);
+            } else {
+                PRINTDEBUGMESSAGE("Unknown command given, exiting.", timestamp, type, bank);
+                exit(-1);
+            }
+        }
+        else PRINTDEBUGMESSAGE("Command given to non-existent bank", timestamp, type, bank);
+    }
+} // Counters::evaluateCommands
+
+//call the clear counters
+void DRAMPowerDDR4::clearCountersWrapper()
+{
+    counters.clear();
+}
+
+//////////////////POWER CALCULATION////////////////////////
+
+
+void DRAMPowerDDR4::Energy::clearEnergy(int64_t nbrofBanks) {
+
+    act_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    pre_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    read_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    write_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    ref_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    refb_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    act_stdby_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    pre_stdby_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    idle_energy_act_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    idle_energy_pre_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    f_act_pd_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    f_pre_pd_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    s_pre_pd_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    ref_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    sref_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    sref_ref_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    sref_ref_act_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    sref_ref_pre_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    spup_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    spup_ref_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    spup_ref_act_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    spup_ref_pre_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    pup_act_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    pup_pre_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    total_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+    window_energy_banks.assign(static_cast<size_t>(nbrofBanks), 0.0);
+
+    read_io_energy      = 0.0;
+    write_term_energy   = 0.0;
+    io_term_energy      = 0.0;
+}
+
+void DRAMPowerDDR4::bankEnergyCalc(unsigned vdd)
```

Comment:
please use directly the right mps as input, such thatt the calling function can pass this as argument: memSpec.memPowerSpec[vdd];